tun/tun_linux.go

```diff
@@ -399,6 +491,50 @@ func (tun *NativeTun) Close() error {
 	return err2
 }
 
+func (tun *NativeTun) BatchSize() int {
+	return tun.batchSize
+}
+
+const (
+	// TODO: support TSO with ECN bits
```

Comment:
What's the status of this TODO? How important is it? Any plans to do it? Or not really?

Reply:
It's worth noting that the kernel implementation does leak the ECN bits in accordance with some RFC, which helps CAKE do its thing. This has long been an overdue feature of wireguard-go.

Reply:
Noted. In that case I imagine TSO_ECN support should be added before ECN leak. We can certainly add TSO_ECN support in a followup series. I'm not yet familiar with everything required for the ECN leak.

Reply:
Take a look at `PACKET_CB(skb)->ds = ip_tunnel_ecn_encap(0, ip_hdr(skb), skb);` in [send.c](https://git.zx2c4.com/wireguard-linux/tree/drivers/net/wireguard/send.c#n368) and then at `INET_ECN_decapsulate(skb, PACKET_CB(skb)->ds, ip_hdr(skb)->tos);` and `INET_ECN_decapsulate(skb, PACKET_CB(skb)->ds, ipv6_get_dsfield(ipv6_hdr(skb)));` in [receive.c](https://git.zx2c4.com/wireguard-linux/tree/drivers/net/wireguard/receive.c#n384). The `ip_tunnel_ecn_encap` and `INET_ECN_decapsulate` and `ipv6_get_dsfield` functions are all worth looking at in their own right.